app/workers/flickr_photos_importer.rb

```diff
@@ -94,9 +96,12 @@ def flickr_url(owner, flickr_id)
   end
 
   def last_uploaded_ts(photos)
-    photos.to_a.last.dateupload.to_i
-  rescue StandardError => e
-    Rails.logger.warn("Trouble getting oldest upload date from photo #{photos.to_a.last}: #{e}")
+    photo_array = photos.respond_to?(:photos) ? photos.photos.to_a : photos.to_a
```

Comment:
Why are these changes necessary?  The change to the warning also seems more confusing/less helpful/less specific than the current version, while also seeming like a change in functionality?  Ditto the changes starting on ln 56.

Reply:
@lorawoodford  - As for these updates ... I could seem to escape the changes here (especially the mocks) after the upgrade and rspec-rails version upgrade.